### ISSUE TITLE
Using s_random: TypeError: can only concatenate str (not "bytes") to str

### ISSUE THREAD

**Issue by kleest** [jtpereyda/boofuzz#290]
Version: 0.1.5, master (0e48caa4e022e6e35908a19bdb988fec31b98e64)
Python: 3.7

```python
session = Session(target=Target(connection=SocketConnection("127.0.0.1", 8021, proto='tcp')))

s_initialize("t")
s_random(0, 1, 1, name="random")
session.connect(s_get("t"))

session.fuzz()
```

fails with the following backtrace:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    session.fuzz()
  File "/boofuzz/boofuzz/sessions.py", line 598, in fuzz
    self._main_fuzz_loop(self._iterate_protocol())
  File "/boofuzz/boofuzz/sessions.py", line 690, in _main_fuzz_loop
    for fuzz_args in fuzz_case_iterator:
  File "/boofuzz/boofuzz/sessions.py", line 1227, in _iterate_protocol
    for x in self._iterate_protocol_recursive(this_node=self.root, path=[]):
  File "/boofuzz/boofuzz/sessions.py", line 1250, in _iterate_protocol_recursive
    for x in self._iterate_single_node(path):
  File "/boofuzz/boofuzz/sessions.py", line 1273, in _iterate_single_node
    while self.fuzz_node.mutate():
  File "/boofuzz/boofuzz/blocks/request.py", line 78, in mutate
    if item.fuzzable and item.mutate():
  File "/boofuzz/boofuzz/primitives/random_data.py", line 77, in mutate
    self._value += six.int2byte(random.randint(0, 255))
TypeError: can only concatenate str (not "bytes") to str
```
Changing `value` of `s_random` to `str` or `bytes` type fails as well.

**Comment by cq674350529** [comment]
@kleest @jtpereyda This seems to be an issue the implementation of `s_random()`.

https://github.com/jtpereyda/boofuzz/blob/0e48caa4e022e6e35908a19bdb988fec31b98e64/boofuzz/primitives/random_data.py#L10-L37

First, it wraps the value with `str()`. 

https://github.com/jtpereyda/boofuzz/blob/0e48caa4e022e6e35908a19bdb988fec31b98e64/boofuzz/primitives/random_data.py#L74-L78

 Then, in the `mutate()`, `self._value` is initialized with `""`. However, the `type(six.int2byte())` is `bytes` in python3 env.

**Comment by jtpereyda** [comment]
Fixed by PR #292
